src/Microsoft.TemplateEngine.Cli/InstallationRequest.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+
+namespace Microsoft.TemplateEngine.Cli
+{
+    public struct InstallationRequest : IEquatable<InstallationRequest>
+    {
+        public InstallationRequest(string installString, bool isPartOfAnOptionalWorkload = false)
+        {
+            InstallString = installString;
+            IsPartOfAnOptionalWorkload = isPartOfAnOptionalWorkload;
+        }
+
+        /// <summary>
+        /// String to identify the package/directory/archive containing the templates
+        /// </summary>
+        public string InstallString { get; set; }
+
+        /// <summary>
+        /// Is this package being installed as part of an optional workload?
+        /// </summary>
+        public bool IsPartOfAnOptionalWorkload { get; set; }
+
+        public bool Equals(InstallationRequest other)
+        {
+            return InstallString == other.InstallString &&
+                IsPartOfAnOptionalWorkload == other.IsPartOfAnOptionalWorkload;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is InstallationRequest other))
+            {
+                return false;
+            }
+
+            return Equals(other);
+        }
+
+        public override string ToString()
+        {
+            return $"{(IsPartOfAnOptionalWorkload ? "[OW]" : string.Empty)}{InstallString}";
+        }
+
+        public override int GetHashCode()
+        {
+            return unchecked((InstallString?.GetHashCode() ?? 0) + 13 * IsPartOfAnOptionalWorkload.GetHashCode());
```

Comment:
nit: I'm assuming that `HashCode.Combine()` cannot be used because we also target frameworks where this API is not available?

Reply:
`HashCode` is available in `.Net Standard 2.1` while we target `.Net Standard 2.0`. However, there is an official nuget package that makes it available in `2.0` : https://www.nuget.org/packages/Microsoft.Bcl.HashCode

Is it worth adding this new dependency or should we just let it be?

Reply:
I'm fine with leaving it as is, thanks!